source/Calamari/ArgoCD/Helm/HelmValuesFileUpdateTargetParser.cs

```diff
@@ -11,187 +12,87 @@ namespace Calamari.ArgoCD.Helm
 
-        HelmValuesFileImageUpdateTarget? ProcessInlineValuesFile(HelmSource source, string file)
+        (IReadOnlyCollection<HelmValuesFileImageUpdateTarget> Targets, IReadOnlyCollection<HelmSourceConfigurationProblem> Problems) ExtractValuesFilesForSource(HelmSource source)
         {
-            // Check if there is an unaliased annotation
-            var definedPathsForSource = imageReplacePathAnnotations
-                                        .FirstOrDefault(a => a.Key == $"{ArgoCDConstants.Annotations.OctopusImageReplacementPathsKey}")
-                                        .Value;
-            if (definedPathsForSource != null)
-            {
-                if (source.Helm.ValueFiles.Count == 1)
-                {
-                    return new HelmValuesFileImageUpdateTarget(appName,
-                                                               source.Name?.ToApplicationSourceName(),
-                                                               defaultRegistry,
-                                                               source.Path,
-                                                               source.RepoUrl,
-                                                               source.TargetRevision,
-                                                               file,
-                                                               ConvertAnnotationToList(definedPathsForSource));
-                }
-                var valueFilesPathsString = string.Join(", ", source.GenerateValuesFilePaths());
-                throw new InvalidHelmImageReplaceAnnotationsException($"Cannot use {ArgoCDConstants.Annotations.OctopusImageReplacementPathsKey} without an alias when multiple inline values files are present.\n Values Files: {valueFilesPathsString}");
-            }
-
-            // Check for aliased annotation for file
-            var aliasKeyForFile = aliasAnnotations.FirstOrDefault(a => a.Value == file).Key;
-            string? alias = null;
-            if (aliasKeyForFile != null)
-            {
-                alias = GetSpecifierFromKey(aliasKeyForFile);
-            }
-            else
+            var definedPathsForSource = ScopingAnnotationReader.GetImageReplacePathsForApplicationSource(source.Name.ToApplicationSourceName(), annotations, containsMultipleSources);
+            if (!definedPathsForSource.Any())
             {
-                // Check if there is an alias with an absolute path value
-                var absoluteAliasPath = source.GenerateInlineValuesAbsolutePath(file);
-                var aliasKeyForSource = aliasAnnotations.FirstOrDefault(a => a.Value == absoluteAliasPath).Key;
-                if (aliasKeyForSource != null)
+                return (Array.Empty<HelmValuesFileImageUpdateTarget>(), new []
                 {
-                    alias = GetSpecifierFromKey(aliasKeyForSource);
-                }
+                    new HelmSourceIsMissingImagePathAnnotation(source.Name, source.RepoUrl)
```

Comment:
Always return this "problem", it's up to the caller to decide whether it's worth logging a warning or not.

This way `HelmValuesFileUpdateTargetParser` doesn't need to know about scoping annotations